README.md

```diff
@@ -29,23 +29,153 @@ This proposal reached Stage 1 at the 2020 Feb TC39 meeting.
 
 ## Motivation
 
-* Users need all types of duration format depending on the requirements of their application. For example, in the airlines websites, to show how long a flight takes, the duration should be in Short or Narrow format
+* Users need all types of duration format depending on the requirements of their application. For example, to show how long a flight takes, the duration should be in Short or Narrow format
   - 1 hr 40 min 60 sec → Short 
   - 1 h 40 m 60 sec  → Narrow
 
-* Furthermore, for TTS (text to speech) applications, Wide format is needed
-  - 1 hour, 40 minutes and 60 seconds.
-
+## Requirements & Design
+
+In this section, we are going to illustrate each user needs (requirements) and a design for each need (requirement)
+
+### Input Value
+  - Users needs to identify how to input a duration value. For example, if a user needs to format `1000 seconds`, how could the user pass the value to the formatting function.
+
+#### Design
+  * Input value will be an object of type `Temporal.Duration`
+  * Example:
+     * `const d = new DurationFormat();`
+     * `d.format(new Temporal.Duration.from({hours: 3, minutes: 4});`
+
+
+### Formatting width
+Users want to determine several types of the formatting width as following
+  | Format width  | Example               |
+  | -----------   | -----------           |
+  |  Wide         |1 hour and 50 minutes  |
+  |  Short        |1 hr, 50 min           |
+  |  Narrow       |1h 50 m                |
+  |  Dotted       |1: 50: 00              |
+
+
+#### Design
+  * The user can determine the formatting width using a  parameter `style` and the value of this parameter take a ` string` as following:
+      * `“wide”`
+      * `“short”`
+      * `“narrow”`
+      * `“dotted”`
+
+### Supported Time Fields
+Users need the following fields to be supported
+  * Year
+  * Month
+  * Week
+  * Day
+  * Hour
+  * Minute
+  * Second
+  * Millisecond
+  * Microsecond
+  * Nano Second
+
+#### Design
+We are going to support the following fields as a beginning
+  * Week
+  * Day
+  * Hour
+  * Minute
+  * Second
+  * Millisecond
+  * Microsecond
+  * Nano Second
+
+### Determining Time Fields
+Users need to determine the time fields,
+ * For example: 6059 seconds
+   * Hours, Minutes, Seconds
+     * 1 hours, 40 minutes and  59 seconds
+   * Minutes and Seconds
+     * 100 minutes and 59 seconds.
+ 
+#### Design
+   * Users can determine the time fields using `LargestField` and `SmallestField` parameters.
```

Comment:
The first letter should be lowercase.  `largestField` and `smallestField`

Reply:
Good catch !

Done!